### ISSUE TITLE
Problem when using UGRID aggregated in THREDDS

### ISSUE THREAD

**Issue by marceloandrioni** [Reading-eScience-Centre/edal-java#121]
Hello, following @https://github.com/Unidata/thredds/issues/526#issuecomment-287131569 by @drf5n I managed to aggregate unstructured grid results in TDS 4.6.14 as a FMRC (feature model run collection). The solution was basically to add a record dimension to aggregate everything along.
TDS shows the new aggregated dataset just fine using the OPENDAP protocol. All other protocols (NCSS, WMS, WCS) fail, but I think that is expected since only TDS 5 will have true UGRID support.

Back to the main issue, when trying to open the opendap link using ncWMS2/EDAL (version 1.4.1, not the one inside TDS) it fails with the message:
`uk.ac.rdg.resc.edal.exceptions.DataReadingException: Coordinate variables listed in mesh_topology:node_coordinates must share the same single dimension to be UGRID compliant`

Checking the [code](https://github.com/Reading-eScience-Centre/edal-java/blob/0ec0dd7ed3480c56913b91d4a026eaa420636507/cdm/src/main/java/uk/ac/rdg/resc/edal/dataset/cdm/CdmGridDatasetFactory.java#L758) my extremely limited java knowledge tells me that the problem is because now latitude and longitude have one extra dimension (record) besides the main one (node).

[Here](https://github.com/Reading-eScience-Centre/edal-java/files/3611710/files_test.zip) some example of the files I am testing with.

Thank you very much.

**Comment by javirg** [comment]
Please check your files. They are not CF-compliant:
`------------------
Checking variable: latitude
------------------
ERROR: (4.3): Invalid value for positive attribute

------------------
Checking variable: longitude
------------------
ERROR: (4.3): Invalid value for positive attribute
`

**Comment by marceloandrioni** [comment]
Hi @javirg , thank you for your reply. Those error mesages came from EDAL/ncWMS log?

**Comment by javirg** [comment]
No, I just checked your files using https://pumatest.nerc.ac.uk/cgi-bin/cf-checker.pl

**Comment by marceloandrioni** [comment]
Hi @javirg, thanks for pointing me out to [cf-checker](https://github.com/cedadev/cf-checker). I never tried before but it seems a great tool. I will try to fix the errors listed by it to see if then the file can be read/displayed using EDAL.

Thank you.
